# HG changeset patch
# User Rik <rik@octave.org>
# Date 1346820972 25200
#      Tue Sep 04 21:56:12 2012 -0700
# Node ID 704ab1b4c369f28f5f7295375b953945fb1624f0
# Parent  2f0b74cf8b4dd5c7f0750ee689c13c96eb62c05b
build: Properly choose qmake if Qt3 and Qt4 are installed (Bug #37101)

configure.ac: Look for qmake-qt5, qmake-qt4 ahead of qmake.  Verify
output of qmake.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -2191,23 +2191,27 @@ AC_ARG_ENABLE(gui,
 
 if test $build_gui = yes; then
   ## Check for Qt libraries
   PKG_CHECK_MODULES(QT, [QtCore, QtGui, QtNetwork],
     [],
     [AC_MSG_ERROR([Qt libraries are required to build the GUI])])
 
   ## Check for Qt utility programs
-  AC_CHECK_PROGS(QMAKE, [qmake qmake-qt4 qmake-qt5])
+  AC_CHECK_PROGS(QMAKE, [qmake-qt5 qmake-qt4 qmake])
   if test -z "$QMAKE"; then
     AC_MSG_ERROR([qmake is required to build the GUI])
   fi
 
-  QT_INCDIR=`$QMAKE -query | sed -n -e 's/^QT_INSTALL_HEADERS://p' | sed -e 's,\\\\,/,g'`
-  QT_LIBDIR=`$QMAKE -query | sed -n -e 's/^QT_INSTALL_LIBS://p' | sed -e 's,\\\\,/,g'`
+  QT_INCDIR=`$QMAKE -query QT_INSTALL_HEADERS | sed -e 's,\\\\,/,g'`
+  QT_LIBDIR=`$QMAKE -query QT_INSTALL_LIBS | sed -e 's,\\\\,/,g'`
+
+  if test "$QT_INCDIR" = "**Unknown**"; then
+    AC_MSG_ERROR([qmake >= Qt4 is required to build the GUI])
+  fi
 
   AC_CHECK_PROGS(MOC, [moc moc-qt4 moc-qt5])
   AC_CHECK_PROGS(UIC, [uic uic-qt4 uic-qt5])
   AC_CHECK_PROGS(RCC, [rcc])
   if test -z "$MOC" || test -z "$UIC" || test -z "$RCC"; then
     AC_MSG_ERROR([Qt utility programs moc, uic, and rcc are required to build the GUI])
   fi 
 
